packages/solid/web/src/index.ts

```diff
@@ -123,18 +125,32 @@ export type DynamicProps<T extends ValidComponent, P = ComponentProps<T>> = {
  */
 export function Dynamic<T extends ValidComponent>(props: DynamicProps<T>): Accessor<JSX.Element> {
   const [p, others] = splitProps(props, ["component"]);
-  const cached = createMemo<Function | string>(() => p.component);
+  return DynamicComponent(() => p.component, others);
+}
+
+/**
+ * renders an arbitrary custom or native component with props
+ * ```typescript
+ * let Link = DynamicComponent('a', {href:'https://www.solidjs.com/'})
+ * ```
+ * @description https://www.solidjs.com/docs/latest/api#dynamiccomponent
+ */
+export function DynamicComponent<T extends ValidComponent>(
+  theComponent: Function | string,
+  props: DynamicProps<T>
```

Comment:
that's for `web/server/index.ts` ? on `web/src/index.ts` seems to be  `T | undefined`.

Reply:
Both:

https://github.com/solidjs/solid/blob/4dd99a7a590b5baa32175198c6c2013d6584c6fb/packages/solid/web/src/index.ts#L114

https://github.com/solidjs/solid/blob/4dd99a7a590b5baa32175198c6c2013d6584c6fb/packages/solid/web/server/index.ts#L29

I don't know if this function should have generic types because it's for internal using. What I do know is that the `component` property should not be enforced.

Reply:
do you have any suggestion for the types of both functions, I dont do ts.